latex-template/vXXX/Snakefile

```diff
@@ -15,15 +15,18 @@ rule create_plot:
 rule create_pdf:
     input:
         "v{number}.tex",
-        "build",
         "build/plot.pdf"
     output:
         "build/v{number}.pdf"
     shell:
         """
         TEXINPUTS=..: BIBINPUTS=..: max_print_line=1048576 latexmk \
          --lualatex \
-         --output-directory={input[1]} \
+         --output-directory=build \
          --interaction=nonstopmode \
          --halt-on-error {input[0]}
         """
+
+rule clean:
```

Comment:
Is this needed? Maybe this can be prevented
with ` temp(directory("build"))` as output of `rule create_pdf`.
One could also delete all the unnecessary files with the `onsuccess` option maybe?

Reply:
Iam not 100% sure but isnt latexmk 'smart' about which things have to be recompiled and uses these files? If so, deleting the aux files every time would slow done the lualatex runs (even more). 🤔

Reply:
Ah nevermind I could copy the rule from the Snakfile 😅

Does this work?
`
rule create_pdf:
    input: "FORCE"
    output: "build/v{number}.pdf

rule run_latexmk:
    input: "v{number}.tex", "build/plot.pdf" 
    output: "FORCE"
    shell: """ TEXINPUTS=..: BIBINPUTS=..: max_print_line=1048576 latexmk \ --lualatex \ --output-directory=build \ --interaction=nonstopmode \ --halt-on-error {input[0]} """

`

Reply:
> Ah nevermind I could copy the rule from the Snakfile 😅
> 
> Does this work?
> 
> ```
> rule create_pdf:
>     input: "FORCE"
>     output: "build/v{number}.pdf
> 
> rule run_latexmk:
>     input: "v{number}.tex", "build/plot.pdf" 
>     output: "FORCE"
>     shell: """ TEXINPUTS=..: BIBINPUTS=..: max_print_line=1048576 latexmk \ --lualatex \ --output-directory=build \ --interaction=nonstopmode \ --halt-on-error {input[0]} """
> ```

No it does not, Snakemake is to sophisticated 😂

Reply:
Unfortunately not:
`Wildcards in input files cannot be determined from output files:
'number'`

Reply:
> Can you share the code you tried?

I tried working with an actually created FORCE file, and delete it after every latexmk run, which seems dirty to do.

Reply:
```
rule FORCE:
    output: "build/z"
    shell: "touch build/z"

rule run_latexmk:
    input:
        "v{number}.tex",
        "build/plot.pdf",
        "build/z"
    output:
        "build/v{number}.pdf"
    shell:
        """
            TEXINPUTS=..: \
            BIBINPUTS=..: \
            max_print_line=1048576 \
            latexmk \
                --lualatex \
                --output-directory=build \
                --interaction=nonstopmode \
                --halt-on-error \
            {input[0]} \
            && rm build/z
        """
```

Reply:
If we give the file a proper name e.g. `latexmk-finshed` I dont think it is that bad.

Reply:
On that note, does anybody know which infos are used by snakemake to calculate the if a file is 'newer'. I tried to bump the timestamp on the tex file by a minute after the latexmk run (using
`touch -d '+1 minute'`) but snakemake saw through my ruse 😅

Reply:
I played around with it somemore and the solution with the least side effects* 
I could come up with is this:

We colud add a executable file to our template
with the content

```
snakemake --cores 1 --forcerun create_pdf
```
It is a bit more cumbersome to type ./<filename>
but from an other perspective this could be seen as a realistic exable for the usage of executable files: ease of use

One thing I'm not certain about is the name of the exec'able file.

`snakemake`: Seems to be the first contender but in testig I encounter an obvious problem first hand: omitting (aka forgetting) the leading `./` is an easy mistake to make could be a bit confusing. But this mistake is just one of many the participants are going to encounter and it is not a grave mistake anyway.
(Maybe capital `Snakemake` could be the solution to distinguish the file and the program though I dont know if all OS's we 'support' are casesensitive ... macOS for example).



*the profile envvar (see above) would be a global setting that would not be applicable in any other situation ... we would need to use direnv or something along those lines.